iceoryx_hoofs/test/moduletests/test_relative_pointer.cpp

```diff
@@ -308,138 +233,120 @@ TYPED_TEST(relativeptrtests, AssignmentOperatorTests)
+    RelativePointer<TypeParam> rp1(this->memoryPartition[0], 1);
 
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), false);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), false);
     EXPECT_EQ(rp1.unregisterPtr(1), true);
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
 }
 
-TYPED_TEST(relativeptrtests, unRegisterPointerTest_Valid)
+TYPED_TEST(base_relative_ptr_test, unRegisterPointerTest_Valid)
 {
-    MemMap memMap(this->m_fileDescriptor);
-    RelativePointer<TypeParam> rp1(memMap.getMappedAddress(), 1);
+    RelativePointer<TypeParam> rp1(this->memoryPartition[0], 1);
 
-    rp1.registerPtr(1, memMap.getMappedAddress());
+    rp1.registerPtr(1, this->memoryPartition[0]);
     EXPECT_EQ(rp1.unregisterPtr(1), true);
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
 }
 
-TYPED_TEST(relativeptrtests, unregisterPointerAll)
+TYPED_TEST(base_relative_ptr_test, unregisterPointerAll)
 {
-    MemMap memMap(this->m_fileDescriptor);
-    MemMap memMap1(this->m_fileDescriptor);
+    RelativePointer<TypeParam> rp1(this->memoryPartition[0], 1);
+    RelativePointer<TypeParam> rp2(this->memoryPartition[1], 9999);
 
-    RelativePointer<TypeParam> rp1(memMap.getMappedAddress(), 1);
-    RelativePointer<TypeParam> rp2(memMap1.getMappedAddress(), 9999);
-
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
-    EXPECT_EQ(rp2.registerPtr(9999, memMap1.getMappedAddress()), true);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
+    EXPECT_EQ(rp2.registerPtr(9999, this->memoryPartition[1]), true);
     BaseRelativePointer::unregisterAll();
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
-    EXPECT_EQ(rp2.registerPtr(9999, memMap1.getMappedAddress()), true);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
+    EXPECT_EQ(rp2.registerPtr(9999, this->memoryPartition[1]), true);
 }
 
-TYPED_TEST(relativeptrtests, registerPtrWithId)
+TYPED_TEST(base_relative_ptr_test, registerPtrWithId)
 {
-    MemMap memMap(this->m_fileDescriptor);
-    MemMap memMap1(this->m_fileDescriptor);
-
-    RelativePointer<TypeParam> rp1(memMap.getMappedAddress(), 1);
-    RelativePointer<TypeParam> rp2(memMap1.getMappedAddress(), 10000);
+    RelativePointer<TypeParam> rp1(this->memoryPartition[0], 1);
+    RelativePointer<TypeParam> rp2(this->memoryPartition[1], 10000);
 
-    EXPECT_EQ(rp1.registerPtr(1, memMap.getMappedAddress()), true);
-    EXPECT_EQ(rp2.registerPtr(10000, memMap1.getMappedAddress()), false);
+    EXPECT_EQ(rp1.registerPtr(1, this->memoryPartition[0]), true);
+    EXPECT_EQ(rp2.registerPtr(10000, this->memoryPartition[1]), false);
 }
```

Comment:
honestly, I don't understand this test

Reply:
Me neither, but I would leave it as it is so that @MatthiasKillat or someone else can refactor it later.